src/main/java/winstone/HostConfiguration.java

```diff
@@ -250,43 +254,52 @@ protected File getWebRoot(File requestedWebroot, File warfile) throws IOExceptio
             if(!timestampFile.exists() || Math.abs(timestampFile.lastModified()- warfile.lastModified())>1000) {
                 // contents of the target directory is inconsistent from the war.
                 deleteRecursive(unzippedDir);
-                unzippedDir.mkdirs();
+                if (!unzippedDir.mkdirs()) {
+                    Logger.logDirectMessage(Logger.WARNING, null, "Failed to recreate dirs " + unzippedDir.getAbsolutePath(), null);
+                }
             } else {
                 // files are up to date
                 return unzippedDir;
             }
-            
+
             // Iterate through the files
             byte buffer[] = new byte[8192];
-            JarFile warArchive = new JarFile(warfile);
-            for (Enumeration e = warArchive.entries(); e.hasMoreElements();) {
-                JarEntry element = (JarEntry) e.nextElement();
-                if (element.isDirectory()) {
-                    continue;
-                }
-                String elemName = element.getName();
+            try (JarFile warArchive = new JarFile(warfile)) {
+                for (Enumeration<JarEntry> e = warArchive.entries(); e.hasMoreElements();) {
+                    JarEntry element = e.nextElement();
+                    if (element.isDirectory()) {
+                        continue;
+                    }
+                    String elemName = element.getName();
 
-                // If archive date is newer than unzipped file, overwrite
-                File outFile = new File(unzippedDir, elemName);
-                if (outFile.exists() && (outFile.lastModified() > warfile.lastModified())) {
-                    continue;
-                }
-                outFile.getParentFile().mkdirs();
-
-                // Copy out the extracted file
-                try (InputStream inContent = warArchive.getInputStream(element);
-                        OutputStream outStream = new FileOutputStream(outFile)) {
-                    int readBytes = inContent.read( buffer );
-                    while ( readBytes != -1 ) {
-                        outStream.write( buffer, 0, readBytes );
-                        readBytes = inContent.read( buffer );
+                    // If archive date is newer than unzipped file, overwrite
+                    File outFile = new File(unzippedDir, elemName);
+                    if (outFile.exists() && (outFile.lastModified() > warfile.lastModified())) {
+                        continue;
+                    }
+                    try {
+                        Files.createDirectories(Paths.get(outFile.getParent()));
```

Comment:
FWIW in Jenkins core we have been catching `InvalidPathException` as well, which apparently can happen under some obscure conditions on Windows servers.

Reply:
Would recommend to address it before merging